Enforce contiguity for alltoall

xw285cornell opened the pull request:
Summary: We cannot relax the alltoall contiguous requirement which will lead to wrong results.

Test Plan: Added a test.

Differential Revision: D66560930




cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

fduwjj reviewed (commented):
cc: @kwen2501 to take a look since you made the change for P2P ops.

xw285cornell committed:
Enforce contiguity for alltoall (#141816)

Summary:

We cannot relax the alltoall contiguous requirement which will lead to wrong results.

Test Plan: Added a test.

Reviewed By: fduwjj

Differential Revision: D66560930

xw285cornell commented:
I looked (very hard) at the errors and none of them seem relevant. So I'm landing it - if it's breaking anything please feel free to revert.
